core/appStatusPolling/appStatusPooling_test.go

```diff
@@ -0,0 +1,65 @@
+package appStatusPolling_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core/appStatusPolling"
+	"github.com/ElrondNetwork/elrond-go/core/mock"
+	"github.com/ElrondNetwork/elrond-go/statusHandler"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewAppStatusPooling_NilAppStatusHandlerShouldErr(t *testing.T) {
+	_, err := appStatusPolling.NewAppStatusPolling(nil, 10)
+	assert.Equal(t, err, appStatusPolling.NilAppStatusHandler)
+}
+
+func TestNewAppStatusPooling_NegativePollingDurationShouldErr(t *testing.T) {
+	_, err := appStatusPolling.NewAppStatusPolling(&statusHandler.NilStatusHandler{}, -10)
+	assert.Equal(t, err, appStatusPolling.PollingDurationNegative)
+}
+
+func TestNewAppStatusPooling_OkValsShouldPass(t *testing.T) {
+	_, err := appStatusPolling.NewAppStatusPolling(&statusHandler.NilStatusHandler{}, 10)
```

Comment:
might call the constructors with something like time.Millisecond or greater as 10 nanoseconds will lead that this test will call its go routine multiple times until it will finish (cpu time wasted in tests)

Reply:
done